columnphysics/icepack_mechred.F90

```diff
@@ -1825,33 +1826,41 @@ subroutine icepack_step_ridge (dt,           ndtd,         &
       real (kind=dbl_kind) :: &
          dtt          ! thermo time step
 
-      real (kind=dbl_kind), dimension(:), allocatable :: &
-         l_fiso_ocn     ! local isotope flux to ocean  (kg/m^2/s)
-
       real (kind=dbl_kind) :: &
          l_closing      ! local rate of closing due to divergence/shear (1/s)
 
       logical (kind=log_kind) :: &
          l_closing_flag ! flag if closing is passed
 
+      logical (kind=log_kind), save :: &
+         first_call = .true.   ! first call flag
+
       character(len=*),parameter :: subname='(icepack_step_ridge)'
 
+      !-----------------------------------------------------------------
+      ! Check optional arguments
+      !-----------------------------------------------------------------
+
+      if (argcheck == 'always' .or. (argcheck == 'first' .and. first_call)) then
+         if (tr_iso) then
+            if (present(fiso_ocn)) then
+                ! OK
+            else
```

Comment:
switch the logic to `if (.not.(present(tr_iso))) then` ?

Reply:
Maybe.  But if you need 4 optional arguments, I think this syntax is clearer.  For example,

```
if (present(x1) .and. present(x2). and. present(x3).and.present(x4)) then
  !OK
else
  abort
endif
```

rather than

```
if (.not.present(x1) .or..not.present(x2).or..not.present(x3).or..not.present(x4)) then
  abort
endif
```

I prefer the former.  I know it's awkward too but I think it's cleaner.  I guess a question is whether

```
if (.not.(present(x1).and.present(x2).and.present(x3).and.present(x4))) then
  abort
endif
```

this would also work and maybe be best?  Happy for some input.

Reply:
I ended up updating it using the final option above.